### PR TITLE
Add label_from_zero_one argument to LogisticLoss

### PR DESCRIPTION
**Pull request by sxjscience**
## Description ##
Add the `use_zero_one` argument to LogisticLoss. This handles the case where the labels are either 0 or 1.

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] use_zero_one flag

### CODE REVIEW

**Comment by piiswrong** [comment]
I think its better to let user do this manually. Options like this are hard to find

**Comment by sxjscience** [comment]
@piiswrong I think we should add this flag. It's very common that the labels we use are 0/1 instead of -1, 1.

**Comment by ZiyueHuang** [comment]
Agree that 0/1 label is more common. And it seems that `LogisticRegressionOutput` in symbol only support 0/1 label.

**Comment by sxjscience** [comment]
There are other such kind of options in the our current loss APIs, e.g, `from_logits` in SoftmaxCrossEntropyLoss (http://mxnet.incubator.apache.org/api/python/gluon/loss.html#mxnet.gluon.loss.SoftmaxCrossEntropyLoss). Users can rely on the document to find these options.

**Comment by piiswrong** [comment]
use_zero_one is a bad name.
an alternative is to explain how to transform 0 1 label to -1 1 in comment

**Comment by sxjscience** [comment]
how about changing the name to `label_zero_one`?

**Comment by sxjscience** [comment]
or change it to `labal_from_zero_one`

**Comment by chinakook** [comment]
+1 and -1 label is very convenient because the loss and the backward can be calculated in a more mathematical elegant way. 
However, the autograd mechanism of gluon implementation cannot get benefit from that because the backward can be manually(tricky) calculated as "grad = -label / (1 + exp(pred * label))" which is more numerical stable.
So, I think +1 and -1 label may not have much effect here.

**Comment by piiswrong** [comment]
how about label_format = 'signed'/'binary'

**Comment by sxjscience** [comment]
Sounds good

**Comment by sxjscience** [comment]
@piiswrong I've renamed accordingly.

File: python/mxnet/gluon/loss.py [hunk 1]
```diff
@@ -619,16 +619,19 @@ class LogisticLoss(Loss):
...
     batch_axis : int, default 0
         The axis that represents mini-batch.
-
+    label_format : str, default 'signed'
+        Can be either 'signed' or 'binary'. If the label_format is 'signed', all label values should
+        be either -1 or 1. If the label_format is 'binary', all label values should be either
+        0 or 1.
 
     Inputs:
         - **pred**: prediction tensor with arbitrary shape.
```
Comment (szha) [inline python/mxnet/gluon/loss.py]: The docstring below should have been updated to reflect the introduction of `label_format`
